### ISSUE TITLE
Question on Loss Calculation

### ISSUE THREAD

**Issue by Morris135212** [tloen/alpaca-lora#206]
When I see closely on the Loss calculation, I found something weird, the input and the labels share the exact same ids
- So the first question: is it design on purpose?
<img width="999" alt="截屏2023-03-29 11 44 11" src="https://user-images.githubusercontent.com/91228632/228422342-85962ff6-de2b-4f9a-abb7-17b644652b26.png">

If it is, then another weird thing is about the loss calculation. In the code, the loss calculation is like

```
shift_logits = logits[..., :-1, :].contiguous()
shift_labels = labels[..., 1:].contiguous()
loss_fct = CrossEntropyLoss()
loss = loss_fct(shift_logits.view(-1, self.config.vocab_size), shift_labels.view(-1))
```

This doesn't make any sense to me if labels share the same ids with the input. If it is, then it just like input the original context, and predict the original context. Does it really help when finetuning a model for generation purpose?
It just doesn't like how it is trained on other large language model where in the input part, the label context is masked, and the model just need to predict and generate the label part.
Really confused now. Can someone be nice to answer my question? Thanks a lot!!!

**Comment by tloen** [comment]
The LLM prediction task is to predict inputs[i] from inputs[:i]. The labels are the same as the inputs, and are shifted in the loss calculation.
